lib/src/shared/codec.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:archive/archive.dart';
+
+abstract class Codec {
+  /// Returns the message encoding that this compressor uses.
+  ///
+  /// This can be values such as "gzip", "deflate", "snappy", etc.
+  String messageEncoding();
+
+  /// Wraps an existing output stream with a compressed output.
+  List<int> compress(List<int> outputStream);
+
+  /// Wraps an existing output stream with a uncompressed input data.
+  List<int> decompress(List<int> inputStream);
+}
+
+/// The "identity", or "none" codec.
+///
+/// This codec is special in that it can be used to explicitly disable Call
+/// compression on a Channel that by default compresses.
+class Identity implements Codec {
+  const Identity();
+
+  @override
+  String messageEncoding() {
+    return "identity";
+  }
+
+  @override
+  List<int> compress(List<int> outputStream) {
```

Comment:
I would not call this a `outputStream` - this is a bit confusing (because it is not a `Stream`). I think better name is just `data`.